xbmc/music/windows/GUIWindowVisualisation.cpp

```diff
@@ -240,3 +241,29 @@ void CGUIWindowVisualisation::FrameMove()
   }
   CGUIWindow::FrameMove();
 }
+
+void CGUIWindowVisualisation::ToggleOSD()
+{
+  CGUIDialogMusicOSD *pOSD = static_cast<CGUIDialogMusicOSD *>(g_windowManager.GetWindow(WINDOW_DIALOG_MUSIC_OSD));
+  if (pOSD)
+  {
+    if (pOSD->IsDialogRunning())
+    {
+      pOSD->Close();
```

Comment:
If you open the OSD with ACTION_SHOW_OSD this window lost the focus and the OSD will consume all messages. IMO this method is never called in case the the OSD is running.